app/services/imports/source_detector.rb

```
@@ -0,0 +1,235 @@
+  end
+
+  private
+
+  attr_reader :file_content, :filename, :file_path
+
+  def gpx_file?
+    return false unless filename
+
+    # Must have .gpx extension AND contain GPX XML structure
+    return false unless filename.downcase.end_with?('.gpx')
+    
```

RuboCop (1.76.1)

[convention] Trailing whitespace detected.

(Layout/TrailingWhitespace)